src/noggit/texture_set.cpp

```diff
@@ -425,11 +426,40 @@ bool TextureSet::paintTexture(float xbase, float zbase, float x, float z, Brush*
             else
             {
               amaps[layer][offset] -= alpha_change * (amaps[layer][offset] / sum_other_alphas);
+
+            }
+
+            if (amaps[layer][offset] > 1.0f)
+                amaps[layer][offset] = 1.0f;
+            else if (amaps[layer][offset] < 0.0f || isnan(amaps[layer][offset]))
+                    amaps[layer][offset] = 0.0f;
+
+            if(layer != 0)
+                totOthers += amaps[layer][offset];
+          }
+          if (totOthers > 1)
+          {
+            float mul = (1 / totOthers);
+
+            amaps[1][offset] = amaps[1][offset] * mul;
```

Comment:
This one isn't guarded, but probably is implicitly guarded by the condition only happening if `nTextures > 0`. The loop

```
auto corrected_totOthers = 0.f;
for (auto layer (1); layer < nTextures; ++layer) { 
  amaps[layer][offset] = amaps[layer][offset] / totOthers;
  corrected_totOthers += amaps[layer][offset];
}
amaps[0][offset] = std::max(0.0f, std::min(1.0f - corrected_totOthers, 1.0f));
```
should be equivalent?